src/let-syntax/lexer.mll

```diff
@@ -28,12 +28,21 @@ let fail lb msg =
   exit 1
 
 let ps t s =
-  String.iter s ~f:(function
-    | '\n' -> t.line <- t.line + 1
-    | _    -> ());
-  output_string t.oc s
+  let len = String.length s in
+  if len > 0 then
+    if s.[len - 1] = '\n' then begin
+      assert (String.index s '\n' = len - 1);
+      t.line <- t.line + 1;
+      if len > 1 && s.[len - 2] = '\r' then begin
+        output_substring t.oc s 0 (len - 2);
+        output_char t.oc '\n'
+      end else
```

Comment:
Note that this is not equivalent to the old code as we were scanning the whole string

Reply:
Ah, indeed